### ISSUE TITLE
make pmap docs point more clearly to collectives

### ISSUE THREAD

**Issue by mattjj** [google/jax#3430]
At the top level of jax.readthedocs.io, we have a section on `pmap` titled "Parallelism", but that doesn't include any obvious link to the documentation on collectives in `lax`, e.g. `lax.axis_index`.

**Comment by 8bitmp3** [comment]
Hi @mattjj, do you mean there's no obvious reference to `lax` > "`axis_index` (axis_name): Return the index along the pmap**ped axis `axis_name`" (emphasis added)? In Sphinx's doc builder, there's a tables of contents tree—a `toctree` with links to subpages and other content. So, perhaps it would require changing `pmap`'s `toctree` in [`jax.rst`](https://github.com/google/jax/blob/master/docs/jax.rst)  by manually editing the `toctree` called `autosummary` (generated by this [extension](https://sphinx-doc-zh.readthedocs.io/en/latest/ext/autosummary.html) from Python funcs, etc) with a link to `jax.lax.axis_index`(which lives under **Public API: jax package** > **Subpackages**)? Cheers 👍
 
<img width="321" alt="image" src="https://user-images.githubusercontent.com/19637339/84697882-ce2f4f80-af46-11ea-9e6f-44b2733c8b6e.png">

**Comment by mattjj** [comment]
We made [the shard_map docs](https://jax.readthedocs.io/en/latest/notebooks/shard_map.html) talk about collectives a lot!
